docs/udf.rst

```diff
@@ -0,0 +1,367 @@
+.. _ksql-udfs:
+
+KSQL User Defined Function Reference
+====================================
+
+KSQL supports creating User Defined Functions (UDFs) via custom jars that are
```

Comment:
Shouldn't we introduce the terms UDF and UDAFs explicitly?  I wouldn't lump them together.  Otherwise it's e.g. confusing why one has an annotation called `@UdafFactory`.

Reply:
See this example juxtaposition. We might want to include something similar here in table format:
<img width="644" alt="screen" src="https://user-images.githubusercontent.com/294849/42177069-9815d228-7e2b-11e8-8483-be813f4e9b42.png">